### PR TITLE
Non module support

### PR DESCRIPTION
**Pull request by gauntface**
This PR changes how Typescript is built for the client.

Instead of using just the typescript compiler, this runs the client typescript files through rollup with a typescript plugin to handle transpilation of TS to ES2015.

- This will build all files in `src/client/`. Rollup is generally designed to have "entry" points which we can use in the future if we want, but for now, this seems like most flexible - just build everything as an entry point.
- Also fix the breaking change I made to the dashbaord.

### CODE REVIEW

File: gulp-tasks/build-client.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 const path = require('path');
-const {buildTypescript} = require('./utils/build-typescript');
+const {buildBrowserTypescript} = require('./utils/build-browser-typescript');
 
 function buildClient() {
-  const destDir = path.join(global.__buildConfig.dest, 'client')
```
Comment (gauntface) [inline gulp-tasks/build-client.js]: Adding my own comment to gather test data.

**Comment by samuelli** [comment]
LGTM, but deferring to @aomarks. PTAL.

File: gulp-tasks/utils/build-browser-typescript.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    input: scriptPath,
+    output: {
+      format: 'iife',
+      sourcemap: true,
+      name: path.basename(scriptPath),
+    },
+    plugins: [
+      typescriptPlugin({
+        ...tsconfig.compilerOptions,
+        tsconfig: false,
```
Comment (aomarks) [inline gulp-tasks/utils/build-browser-typescript.js]: This ignores options in the tsconfig -- that doesn't seem right?

https://github.com/rollup/rollup-plugin-typescript#usage

Comment (aomarks) [reply gulp-tasks/utils/build-browser-typescript.js]: Oh sorry ignore me, missed the line right above.

**Pull request merged**